wagtail/snippets/views/snippets.py

```diff
@@ -502,17 +514,48 @@ def redirect_to_usage(request, app_label, model_name, pk):
     )
 
 
-class HistoryView(IndexView):
-    history_url_name = None
-    template_name = "wagtailadmin/generic/index.html"
-    page_title = gettext_lazy("Snippet history")
+class SnippetHistoryReportFilterSet(WagtailFilterSet):
+    action = django_filters.ChoiceFilter(choices=log_registry.get_choices)
+    user = django_filters.ModelChoiceFilter(
+        field_name="user",
+        queryset=lambda request: ModelLogEntry.objects.all().get_users(),
+    )
+    timestamp = django_filters.DateFromToRangeFilter(
+        label=_("Date"), widget=DateRangePickerWidget
+    )
+
+    class Meta:
+        model = ModelLogEntry
+        fields = ["action", "user", "timestamp"]
+
+
+class ActionColumn(Column):
+    cell_template_name = "wagtailsnippets/snippets/revisions/_actions.html"
+
+    def __init__(self, *args, object=None, view=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.object = object
+        self.view = view
+
+    def get_cell_context_data(self, instance, parent_context):
+        context = super().get_cell_context_data(instance, parent_context)
+        context["revision_enabled"] = isinstance(self.object, RevisionMixin)
+        context["object"] = self.object
+        context["view"] = self.view
```

Comment:
Passing the `view` here so that we can use things like `{% url view.revisions_revert_url_name %}` in the template. It needs to be explicitly passed because the components framework doesn't automatically pass context from the view.